python/dgl/sparse/elementwise_op.py

```diff
@@ -85,13 +85,113 @@ def sub(A: Union[DiagMatrix], B: Union[DiagMatrix]) -> Union[DiagMatrix]:
     --------
     >>> A = diag(torch.arange(1, 4))
     >>> B = diag(torch.arange(10, 13))
-    >>> A - B
+    >>> sub(A, B)
     DiagMatrix(val=tensor([-9, -9, -9]),
     shape=(3, 3))
     """
     return A - B
 
 
+def mul(
+    A: Union[SparseMatrix, DiagMatrix, float, int],
+    B: Union[SparseMatrix, DiagMatrix, float, int],
+) -> Union[SparseMatrix, DiagMatrix]:
+    """Elementwise multiplication for `DiagMatrix` and `SparseMatrix`.
+
+    The supported combinations are shown as follows.
+
+    +--------------+------------+--------------+--------+
+    |    A \\ B    | DiagMatrix | SparseMatrix | scalar |
+    +--------------+------------+--------------+--------+
+    |  DiagMatrix  |     ✅     |      🚫      |   ✅   |
+    +--------------+------------+--------------+--------+
+    | SparseMatrix |     🚫     |      🚫      |   ✅   |
+    +--------------+------------+--------------+--------+
+    |    scalar    |     ✅     |      ✅      |   🚫   |
+    +--------------+------------+--------------+--------+
+
+    Parameters
+    ----------
+    A : SparseMatrix or DiagMatrix or float or int
+        Sparse matrix or diagonal matrix or scalar value
+    B : SparseMatrix or DiagMatrix or float or int
+        Sparse matrix or diagonal matrix or scalar value
+
+    Returns
+    -------
+    SparseMatrix or DiagMatrix
+        Either sparse matrix or diagonal matrix
+
+    Examples
+    --------
+    >>> row = torch.tensor([1, 0, 2])
+    >>> col = torch.tensor([0, 3, 2])
+    >>> val = torch.tensor([10, 20, 30])
+    >>> A = from_coo(row, col, val)
+    >>> mul(A, 2)
+    SparseMatrix(indices=tensor([[1, 0, 2],
+                                 [0, 3, 2]]),
+    values=tensor([20, 40, 60]),
+    shape=(3, 4), nnz=3)
```

Comment:
Is the printing here correct? I remember Minjie improved it before.

Reply:
Nice catch, updated.